scripts/generate_dump

```diff
@@ -290,7 +290,7 @@ main() {
     if [[ $platform == *"mlnx"* ]]; then
         local sai_dump_filename="/tmp/sai_sdk_dump_$(date +"%m_%d_%Y_%I_%M_%p")"
         docker exec -it syncd saisdkdump -f $sai_dump_filename
-        docker cp syncd:$sai_dump_filename /tmp/
+        docker exec syncd tar Ccf $(dirname $sai_dump_filename) - $(basename $sai_dump_filename) | tar Cxf /tmp/ -
```

Comment:
You actually copy the whole /tmp folder to the host. What you need is just one file.

Suggestion:
docker exec -it syncd cat /tmp/dump > /tmp/dump

Reply:
@qiluo-msft , it really only copies the dumpfile "$sai_dump_filename" to the host, not the whole folder, see my test result:


    root@arc-switch1041:/usr/bin# 
    root@arc-switch1041:/usr/bin# ls /tmp/ -l
    total 21640
    -rw-r--r-- 1 root  root       502 Apr 27 00:48 mlnxfwmanager-query.txt
    -rw-r--r-- 1 root  root  22140381 Apr 27 01:00 sai_sdk_dump_04_27_2018_01_00_AM
    -rw-r--r-- 1 admin admin     9548 Apr 26 07:11 sh_dump
    root@arc-switch1041:/usr/bin# docker exec -it syncd ls /tmp -l
    total 21644
    -rw-r--r-- 1 root root        4 Apr 27 00:58 aaa.txt
    -rw-r--r-- 1 root root        4 Apr 27 00:59 bbb.txt
    -rw-r--r-- 1 root root        4 Apr 27 00:59 ccc.txt
    -rw-r--r-- 1 root root        6 Apr 27 00:48 ptf_nn_agent.err.log
    -rw-r--r-- 1 root root        0 Apr 27 00:48 ptf_nn_agent.out.log
    -rw-r--r-- 1 root root       82 Apr 27 00:48 sai.profile
    srwxr-xr-x 1 root root        0 Apr 27 00:48 sai_acl_rpc_socket
    -rw-r--r-- 1 root root 22140381 Apr 27 01:00 sai_sdk_dump_04_27_2018_01_00_AM
    -rw-r--r-- 1 root root        0 Apr 27 00:48 sdk_ready
    srwxr-xr-x 1 root root        0 Apr 27 00:48 sx_api
    root@arc-switch1041:/usr/bin# 

as you can see only the dumpfile "sai_sdk_dump_04_27_2018_01_00_AM" was copied to the host.

@pavel-shirshov "docker cp" is not applicable to tmpfs of the container, see the result:

    root@arc-switch1041:/usr/bin# docker cp syncd:/tmp/sai_sdk_dump_04_27_2018_01_00_AM /tmp/sai_sdk_dump_04_27_2018_01_00_AM
    Error response from daemon: lstat /var/lib/docker/aufs/mnt/342693aeb40ab7f44990c19ff2908c02dd6d4a797c13f044fbcc9263c280d07b/tmp/sai_sdk_dump_04_27_2018_01_00_AM: no such file or directory
    root@arc-switch1041:/usr/bin#

Reply:
@keboliu, Interesting, thank you for the information